### DISCUSSION TITLE
StageWebView crashes on M1 Mac / 33.1.1.686

### DISCUSSION THREAD

**Discussion by perholmes** [airsdk/Adobe-Runtime-Support#1422] [General]
StageWebView crashes when we access it. The only two differences between the previous version of our app that worked and the current crash is that the build machine is now an M1 Mac, and the AIR version is 33.1.1.686.

We don't load web content (only file content), and we do have the proper entitlement of com.apple.security.network.client, which some have noted is necessary for a webview. Haven't tested without.

Here are some interesting snippets from the crash report. It's especially puzzling that we appear to be building an ARM-64 native app, which doesn't work. I need to build an Intel-64 app that can run anywhere. I wasn't aware that AIR could build an ARM Mac app, and this may be the cause of the crash.

Any insights would be much appreciated.

`
Code Type:             ARM-64 (Native)
...
Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000038
Exception Codes:       0x0000000000000001, 0x0000000000000038
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace SIGNAL, Code 11 Segmentation fault: 11
Terminating Process:   exc handler [77395]

VM Region Info: 0x38 is not in any region.  Bytes before following region: 4374527944
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      UNUSED SPACE AT START
--->  
      __TEXT                      104be0000-104be8000    [   32K] r-x/r-x SM=COW  ...Shot Designer

Kernel Triage:
VM - pmap_enter failed with resource shortage
VM - pmap_enter failed with resource shortage
VM - pmap_enter failed with resource shortage
VM - pmap_enter failed with resource shortage
VM - pmap_enter failed with resource shortage

Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   Adobe AIR                     	       0x1076aa490 0x106f78000 + 7546000
1   Adobe AIR                     	       0x1076aa510 0x106f78000 + 7546128
2   Adobe AIR                     	       0x1076aa510 0x106f78000 + 7546128
3   Adobe AIR                     	       0x1076aa32c 0x106f78000 + 7545644
4   Adobe AIR                     	       0x1076a8ee8 0x106f78000 + 7540456
`

**Comment by perholmes** [comment]
Update, we've now tested on Intel Macs, where the app runs fine, as does the webview.

This indicates that

* The AIR webview crashes on M1 Macs.
* The ARM64 Code Type in the crash report is false, the app is still compiled as Intel, even on an M1 Mac, or at least is a universal binary, which is a capability we haven't heard about.

it could still be tied to some Monterey issue like new required entitlements. I'll know when we've seen this run on Monterey Intel Macs.

**Reply by ajwfrost** [reply]
Hi

If you create a StageWebView with default arguments (use native = false) then it uses the old WebKit library that's included in the AIR SDK, the same way that it does with HTMLLoader. But, this binary is only available in an x86_64 format, which means that it fails to load when you're running on M1 hardware.

I would expect it to work if you switch to using StageWebView with a 'true' first argument; we're about to make a release that ignores the argument and always uses the native (OS-provided) web view component; and for HTMLLoader this would mean it also uses a native StageWebView behind the scenes.

So if you're able to try it with `new StageWebView(true)` it would be a good first step, and we'll hopefully then eliminate the problem in a forthcoming release.

thanks
